crates/symbolicator/src/services/download/gcs.rs

```diff
@@ -248,24 +255,37 @@ impl GcsDownloader {
                     let stream = response.bytes_stream().map_err(DownloadError::Reqwest);
 
                     super::download_stream(source, stream, destination, timeout).await
+                // If it's a client error, chances are either it's a 404 or it's permission-related.
```

Comment:
just a note for the future. we wanted to surface permission-related errors to the user, but that never went anywhere. just to keep in mind. maybe its easy enough to add now though?

Reply:
i think with these changes it's possible, although i'm unsure how thorough we could be due to a lack of familiarity on my end with our supported services' SDKs. if we're fine with permission-related errors becoming `CacheSpecificError`s on the data cache and `Negative`s on the derived caches, it would be pretty easy to just quickly convert permissions-related errors and start providing some diagnostics.

i'll open up a follow-up PR once there are only 1-2 PRs left in this chain to explore that idea - let me know if what i've mentioned above sounds alarming or undesirable.

Reply:
opened change in #518